packages/kit/src/runtime/client/client.js

```diff
@@ -1284,32 +1273,32 @@ export function create_client({ target, base, trailing_slash }) {
 			});
 		},
 
-		_hydrate: async ({ status, error, node_ids, params, routeId }) => {
+		_hydrate: async ({
+			status,
+			error: original_error, // TODO get rid of this
+			node_ids,
+			params,
+			routeId,
+			data: server_data_nodes,
+			errors: validation_errors
+		}) => {
 			const url = new URL(location.href);
 
 			/** @type {import('./types').NavigationFinished | undefined} */
 			let result;
 
 			try {
-				/**
-				 * @param {string} type
-				 * @param {any} fallback
-				 */
-				const parse = (type, fallback) => {
-					const script = document.querySelector(`script[sveltekit\\:data-type="${type}"]`);
-					return script?.textContent ? JSON.parse(script.textContent) : fallback;
-				};
-				/**
-				 * @type {Array<import('types').ServerDataNode | null>}
-				 * On initial navigation, this will only consist of data nodes or `null`.
-				 * A possible error is passed through the `error` property, in which case
-				 * the last entry of `node_ids` is an error page and the last entry of
-				 * `server_data_nodes` is `null`.
-				 */
-				const server_data_nodes = parse('server_data', []);
-				const validation_errors = parse('validation_errors', undefined);
-
 				const branch_promises = node_ids.map(async (n, i) => {
+					const server_data_node = server_data_nodes[i];
+
+					if (server_data_node?.type === 'error') {
+						if (server_data_node.httperror) {
+							throw error(server_data_node.httperror.status, server_data_node.httperror.message);
+						}
+
+						throw server_data_node.error;
+					}
```

Comment:
I'm not this has the desired outcome. If there's a server error from loading data we want to show the error page within the correct layout, and this would result in the root error page being shown, if I understand this code correctly.

Reply:
Ah yeah, you're right. Will fix this

Reply:
I realised that errors won't be serialized anyway 🤦 Removed this block.

If a client-side error occurs during `load` then it will fall back to the root error page, which isn't 'correct' but would be finicky to solve. On the basis that this isn't a regression, I think we should gloss over it for now and merge this

Reply:
I didn't mean that we should handle the "client side error occurs during hydration load" error in some way, only that the `throw` case would be wrong, since in case of an error you have the nodes leading up to an error page already and only want to hydrate that, not throw an error along the way. So all good now 👍